Automattic/wp-calypso
# Add/pricing page i5 layout

### Changes proposed in this Pull Request

This PR implements the layout of pricing page i5.

Fixes 1196341175636977-as-1199149328812571

### Implementation notes

- This PR does not update the product cards
- Styles on mobile and Calypso will be updated later in a different PR
- Fixed several TS errors
- Created a new grid component `products-grid-i5`
- Rendered the filter bar conditionally in `SelectorAlt`
- Added placeholders for filter bar, "more" section, and comparison table

### Testing instructions

- Download the PR
- Run both Calypso and Jetpack cloud concurrently
- Select the `i5 - Saas table design` variant of the a/b test `jetpackConversionRateOptimization`
- Visit the _Plans_ page
- In both platform, check:
    - That you see the 3 sections with their headings: _Product Bundles_, _Individual products_, and _Bundle Comparison_ (empty)
    - That you don't see the filter bar, for now (it'll be included in the _Product Bundles_ section)
    - That plans follow this order: Complete, Security Real Time, Security Daily
    - That products follow this order: Backup RT, Backup Daily, Scan, Anti-spam, Site Search, CRM Entrepeneur
- Select every other variants (v0, v1, and v2) and check that the page looks like it does in production (or check captures below)

### Screenshots

#### i5
<img width="1107" alt="Screen Shot 2020-11-10 at 5 15 17 PM" src="https://user-images.githubusercontent.com/1620183/98739993-568c1200-2378-11eb-85b9-14bfe38506ff.png">

#### v2
<img width="1086" alt="Screen Shot 2020-11-10 at 5 20 08 PM" src="https://user-images.githubusercontent.com/1620183/98740428-05305280-2379-11eb-82d2-45f8b903b343.png">


#### v1
<img width="1075" alt="Screen Shot 2020-11-10 at 5 19 32 PM" src="https://user-images.githubusercontent.com/1620183/98740372-f21d8280-2378-11eb-9ed9-2fa9abfa1e2e.png">


#### v0
<img width="1079" alt="Screen Shot 2020-11-10 at 5 18 16 PM" src="https://user-images.githubusercontent.com/1620183/98740295-cbf7e280-2378-11eb-9b01-ad925bf551b1.png">
